File: Java/NetBeans/SV/src/Exe_DSSV/SV.java
package Exe_DSSV;

public abstract class SV {

    protected String hoTen;
    protected int namSinh;
    protected double dtb;

    public SV(String hoTen, int namSinh, double dtb) {
        this.hoTen = hoTen;
        this.namSinh = namSinh;
        this.dtb = dtb;
    }

    public void hienThi() {
        System.out.println(this);
    }

    public boolean equals(Object sv) {
        if (sv != null && sv instanceof SV b) {
            return this.hoTen.equals(b.hoTen)
                    && this.namSinh == b.namSinh
                    && this.dtb == b.dtb;
        }
        return false;
    }

    public String toString() {
        return "Ho ten: " + hoTen + "\n"
                + "Nam sinh: " + namSinh + "\n"
                + "Diem trung binh: " + dtb;
    }

    public abstract String loaiSV();

    public abstract boolean duocTN();
}

File: C#/QLSVTHvaSP/QLSVTHvaSP/SV.cs
namespace QLSVTHvaSP
{
    internal abstract class SV
    {
        protected string HoTen;
        protected int NamSinh;
        protected float DiemTB;

        public string HoTenGS { get => HoTen; set => HoTen = value; }
        public int NamSinhGS { get => NamSinh; set => NamSinh = value; }
        public float DiemTBGS { get => DiemTB; set => DiemTB = value; }

        public SV() { }

        public SV(string hoTen, int namSinh, float diemTB)
        {
            HoTen = hoTen;
            NamSinh = namSinh;
            DiemTB = diemTB;
        }

        public virtual void NhapSV()
        {
            Console.Write("Nhap ho ten: ");
            HoTen = Console.ReadLine();

        }

        public virtual void HienThi()
        {
            Console.Write($"{HoTen}, {NamSinh}, {DiemTB}");
        }

        public override bool Equals(object obj)
        {
            if (obj is null || !(obj is SV))
            {
                SV sv = obj as SV;
                return HoTen.Equals(sv.HoTen) && NamSinh == sv.NamSinh && DiemTB == sv.DiemTB;
            }

            return false;
        }
        abstract public string LoaiSV();
        abstract public bool DuocTN();

        public override int GetHashCode()
        {
            throw new NotImplementedException();
        }
    }
}